dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinderTest.java

```diff
@@ -0,0 +1,264 @@
+
+        DatasetFieldType depositorFieldType = MocksFactory.makeDatasetFieldType("depositor",
+                FieldType.TEXT,
+                false,
+                citationMetadataBlock);
+        depositorFieldType.setDisplayOrder(5);
+
+        DatasetFieldType dateOfDepositFieldType = MocksFactory.makeDatasetFieldType("dateOfDeposit",
+                FieldType.TEXT,
+                false,
+                citationMetadataBlock);
+        dateOfDepositFieldType.setDisplayOrder(6);
+
+        when(datasetFieldService.findByNameOpt(eq("title"))).thenReturn(titleFieldType);
+        when(datasetFieldService.findByNameOpt(eq("author"))).thenReturn(authorFieldType);
+        when(datasetFieldService.findByNameOpt(eq("authorName"))).thenReturn(authorNameFieldType);
+        when(datasetFieldService.findByNameOpt(eq("authorAffiliation"))).thenReturn(authorAffiliationFieldType);
+        when(datasetFieldService.findByNameOpt(eq("datasetContact"))).thenReturn(datasetContactFieldType);
+        when(datasetFieldService.findByNameOpt(eq("datasetContactName"))).thenReturn(datasetContactNameFieldType);
+        when(datasetFieldService.findByNameOpt(eq("datasetContactAffiliation"))).thenReturn(
+                datasetContactAffiliationFieldType);
+        when(datasetFieldService.findByNameOpt(eq("datasetContactEmail"))).thenReturn(datasetContactEmailFieldType);
+        when(datasetFieldService.findByNameOpt(eq("dsDescription"))).thenReturn(dsDescriptionFieldType);
+        when(datasetFieldService.findByNameOpt(eq("dsDescriptionValue"))).thenReturn(dsDescriptionValueFieldType);
+        when(datasetFieldService.findByNameOpt(eq("dsDescriptionDate"))).thenReturn(dsDescriptionDateFieldType);
+        when(datasetFieldService.findByNameOpt(eq("subject"))).thenReturn(subjectFieldType);
+        when(datasetFieldService.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(subjectFieldType,
+                "Chemistry",
+                false))
+                .thenReturn(subjectFieldType.getControlledVocabularyValue("chemistry"));
+        when(datasetFieldService.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(subjectFieldType,
+                "Agricultural Sciences",
+                false))
+                .thenReturn(subjectFieldType.getControlledVocabularyValue("agricultural_sciences"));
+        when(datasetFieldService.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(subjectFieldType,
+                "Arts and Humanities",
+                false))
+                .thenReturn(subjectFieldType.getControlledVocabularyValue("arts_and_humanities"));
+
+        when(datasetFieldService.findByNameOpt(eq("depositor"))).thenReturn(depositorFieldType);
+        when(datasetFieldService.findByNameOpt(eq("dateOfDeposit"))).thenReturn(dateOfDepositFieldType);
+
+
+        dataset = MocksFactory.makeDataset();
+    }
+
+    @Test
+    public void create__WITH_CHANGED_FILE_TERMS() throws IOException, JsonParseException {
```

Comment:
maybe just "getLicenseDifference_...".

Reply:
Tbh all tests have been taken directly from `datasetversiondifferencetest` and I've only limited logic to `LicenseDifference` :)